src/main/java/com/google/sps/servlets/CommentServlet.java

```diff
@@ -136,14 +136,14 @@ public void doDelete(HttpServletRequest request, HttpServletResponse response)
       throws IOException {
     String recipeID = request.getParameter("recipeID");
     String commentID = request.getParameter("commentID");
+    String isLeaf = request.getParameter("isLeaf");
 
-    if (recipeID == null || commentID == null) {
+    if (recipeID == null || commentID == null || isLeaf == null
```

Comment:
I don't think the client should indicate if a comment is a leaf - especially not using a parameter as it can be arbitrarily set by the user. I think deleteComment in FirestoreDB should be doing this check.